contracts/staking/StakingInfo.sol

```diff
@@ -219,11 +226,22 @@ contract StakingInfo {
         emit ReStaked(validatorId, amount, total);
     }
 
-    function logJailed(uint256 validatorId, uint256 exitEpoch)
+    function logJailed(uint256 validatorId, uint256 exitEpoch, address signer)
+        public
+        onlyStakeManager
+    {
+        emit Jailed(validatorId, exitEpoch, signer);
+    }
+
+    function logUnJailed(uint256 validatorId, address signer)
         public
         onlyStakeManager
     {
-        emit Jailed(validatorId, exitEpoch);
+        emit UnJailed(validatorId, signer);
+    }
+
+    function logSlashed(uint256 amount) public onlyStakeManager {
+        emit Slashed(amount);
```

Comment:
Should we also emit who was slashed?

Reply:
no need list of validators is already in `calldata`.